src/Host/Broker/Impl/Interpreters/Interpreter.cs

```diff
@@ -26,13 +27,16 @@ public class Interpreter {
             Version = Version
         };
 
-        public Interpreter(InterpreterManager manager, string id, string name, string path, string binPath, Version version) {
+        public IRInterpreterInfo RInterpreterInfo { get; }
+
+        public Interpreter(InterpreterManager manager, string id, string name, string path, string binPath, Version version, IRInterpreterInfo rInterpreterInfo) {
```

Comment:
If we have `rInterpreterInfo` directly, then a bunch of arguments become redundant, since they're properties of that instance. As do the corresponding properties on this class.